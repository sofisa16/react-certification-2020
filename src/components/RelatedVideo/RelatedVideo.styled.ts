import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';
import {NoStyleLink} from '../../styles/common';

const ParentContainer = styled.div`
  height: 90px;
  position: relative;
  display: flex;
  margin-bottom: 15px;
`;

const RightContainer = styled.img`
  width: 120px;
  height: 90px;
`;

const LeftContainer = styled.div`
  height: 100%;
  -webkit-box-flex: 1;
  flex-grow: 1;
  margin: 0px 10px;
`;

const Title = styled(Typography)`
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const ColorNoStyleLink = styled(NoStyleLink)`
  color: inherit;
`;

const Styled = {
  ParentContainer,
  RightContainer,
  LeftContainer,
  Title,
  ColorNoStyleLink,
};

export default Styled;