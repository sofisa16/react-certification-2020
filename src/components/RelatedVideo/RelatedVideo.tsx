import React from 'react';
import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';

const ParentContainer = styled.div`
  height: 90px;
  position: relative;
  display: flex;
  margin-bottom: 15px;
`;

const RightContainer = styled.img`
  width: 120px;
  height: 90px;
`;

const LeftContainer = styled.div`
  height: 100%;
  -webkit-box-flex: 1;
  flex-grow: 1;
  margin: 0px 10px;
`;

const Title = styled(Typography)`
  display: -webkit-box;
  -webkit-line-clamp: 4;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
`;

interface RelatedVideoProps {
  title: string;
  thumbnails: string;
}

function RelatedVideo(props: RelatedVideoProps): JSX.Element {
  const {title, thumbnails} = props;

  return (
    <ParentContainer>
      <RightContainer src={thumbnails} alt="image" />
      <LeftContainer>
        <Title variant='body1'>
          {title}
        </Title>
      </LeftContainer>
    </ParentContainer>
  );
}

export default RelatedVideo;