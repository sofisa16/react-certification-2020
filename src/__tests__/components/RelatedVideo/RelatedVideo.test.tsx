import React from 'react';
import RelatedVideo from '../../../components/RelatedVideo/RelatedVideo';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders RelatedVideo component', () => {
  const history = createMemoryHistory();
  const route = '/';
  history.push(route);

  const tree = renderer
    .create(
      <Router history={history}>
        <RelatedVideo
          title="title"
          thumbnails="thumbnails"
          videoId="videoId"
          key="key"
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});