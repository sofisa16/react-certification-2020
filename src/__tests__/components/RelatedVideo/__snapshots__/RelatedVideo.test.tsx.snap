// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RelatedVideo component 1`] = `
<a
  className="sc-bdfBwQ sc-jSgupP lclvs hkRxCS"
  href="/videoId"
  onClick={[Function]}
>
  <div
    className="sc-gsTCUz eFWHNA"
  >
    <img
      alt="image"
      className="sc-dlfnbm bPhgq"
      src="thumbnails"
    />
    <div
      className="sc-hKgILt hKROct"
    >
      <p
        className="MuiTypography-root sc-eCssSg iHKEao MuiTypography-body1"
      >
        title
      </p>
    </div>
  </div>
</a>
`;
